modules/adplayerproVideoProvider.js

```diff
@@ -0,0 +1,439 @@
+import {
+  AD_CLICK,
+  AD_COMPLETE,
+  AD_ERROR,
+  AD_IMPRESSION,
+  AD_LOADED,
+  AD_PAUSE,
+  AD_PLAY,
+  AD_REQUEST,
+  AD_SKIPPED,
+  AD_STARTED,
+  DESTROYED,
+  ERROR,
+  MUTE,
+  PLAYER_RESIZE,
+  SETUP_COMPLETE,
+  SETUP_FAILED,
+  VOLUME
+} from '../libraries/video/constants/events.js';
+import {AD_PLAYER_PRO_VENDOR} from '../libraries/video/constants/vendorCodes.js';
+import {getEventHandler} from '../libraries/video/shared/eventHandler.js';
+import {submodule} from '../src/hook.js';
+
+const setupFailMessage = 'Failed to instantiate the player';
+
+/**
+ * @constructor
+ * @param {Object} config - videoProviderConfig
+ * @param {function} adPlayerPro_
+ * @param {CallbackStorage} callbackStorage_
+ * @param {Object} utils
+ * @returns {Object} - VideoProvider
+ */
+export function AdPlayerProProvider(config, adPlayerPro_, callbackStorage_, utils) {
+  const adPlayerPro = adPlayerPro_;
+  let player = null;
+  let playerVersion = null;
+  const playerConfig = config.playerConfig;
+  const divId = config.divId;
+  let callbackStorage = callbackStorage_;
+  let supportedMediaTypes = null;
+  let setupCompleteCallbacks = [];
+  let setupFailedCallbacks = [];
+  const MEDIA_TYPES = [
+    VIDEO_MIME_TYPE.MP4,
+    VIDEO_MIME_TYPE.OGG,
+    VIDEO_MIME_TYPE.WEBM,
+    VIDEO_MIME_TYPE.AAC,
+    VIDEO_MIME_TYPE.HLS
+  ];
+
+  function init() {
+    if (!adPlayerPro) {
+      triggerSetupFailure(-1, setupFailMessage + ': player not present');
+      return;
+    }
+
+    // if (playerVersion < minimumSupportedPlayerVersion) {
+    //   triggerSetupFailure(-2, setupFailMessage + ': player version not supported');
+    //   return;
+    // }
+
+    if (!document.getElementById(divId)) {
+      triggerSetupFailure(-3, setupFailMessage + ': No div found with id ' + divId);
+      return;
+    }
+
+    if (!playerConfig || !playerConfig.placementId) {
+      triggerSetupFailure(-4, setupFailMessage + ': placementId is required in playerConfig');
+      return;
+    }
+
+    triggerSetupComplete();
+  }
+
+  function getId() {
+    return divId;
+  }
+
+  function getOrtbVideo() {
+    supportedMediaTypes = supportedMediaTypes || utils.getSupportedMediaTypes(MEDIA_TYPES);
+
+    const video = {
+      mimes: supportedMediaTypes,
+      protocols: [
+        PROTOCOLS.VAST_2_0,
+        PROTOCOLS.VAST_3_0,
+        PROTOCOLS.VAST_4_0,
+        PROTOCOLS.VAST_2_0_WRAPPER,
+        PROTOCOLS.VAST_3_0_WRAPPER,
+        PROTOCOLS.VAST_4_0_WRAPPER
+      ],
+      // h: player.getHeight(),
+      // w: player.getWidth(),
+      placement: utils.getPlacement(playerConfig.advertising),
+      maxextended: -1, // extension is allowed, and there is no time limit imposed.
+      boxingallowed: 1,
+      playbackmethod: [utils.getPlaybackMethod(config)],
+      playbackend: 1,
+      api: [
+        API_FRAMEWORKS.VPAID_2_0,
+        API_FRAMEWORKS.OMID_1_0
+      ],
+    };
+
+    return video;
+  }
+
+  function getOrtbContent() {
+  }
```

Comment:
If the player has content information such as media source url , media id, media title etc, it would be good to populate. Contextual targeting can help get higher bids

Reply:
This module only supports outstream format, so it does not use content.